go-client/client/tablehandle.go

```diff
@@ -427,22 +428,27 @@ func (th *TableHandle) Count(ctx context.Context, col string) (*TableHandle, err
 // AggBy applies a list of aggregations to table data.
 // See the docs on AggBuilder for details on what each of the aggregation types do.
 func (th *TableHandle) AggBy(ctx context.Context, agg *AggBuilder, by ...string) (*TableHandle, error) {
-	if th.client == nil {
+	if !th.IsValid() {
 		return nil, ErrInvalidTableHandle
 	}
-	return th.client.aggBy(ctx, th, agg, by)
+	return th.client.aggBy(ctx, th, agg.aggs, by)
 }
 
-// Merge combines two or more tables into one aggregate table.
+// Merge combines two or more tables into one table.
 // This essentially appends the tables one on top of the other.
 // If sortBy is provided, the resulting table will be sorted based on that column.
-func (th *TableHandle) Merge(ctx context.Context, sortBy string, others ...*TableHandle) (*TableHandle, error) {
-	if th.client == nil {
-		return nil, ErrInvalidTableHandle
+func Merge(ctx context.Context, sortBy string, tables ...*TableHandle) (*TableHandle, error) {
+	if len(tables) < 1 {
+		return nil, errors.New("must provide at least one table to merge")
 	}
 
-	tables := make([]*TableHandle, len(others)+1)
-	tables[0] = th
-	copy(tables[1:], others)
-	return th.client.merge(ctx, sortBy, tables)
+	for _, table := range tables {
+		if !table.IsValid() {
+			return nil, ErrInvalidTableHandle
+		}
+	}
+
+	client := tables[0].client
```

Comment:
1) Can nil values be passed in?  `null` is totally valid for the server method.  If so, this method needs some changes.

2) What happens if the tables have different clients?

Reply:
1. Maybe. It depends on what we define as `null`. Currently a `nil` `*TableHandle` is treated exactly the same as `&TableHandle{}` or a handle that has been released. However, my first impression is that passing in an invalid table to merge should be an error, while explicitly passing in a `nil` pointer would be okay to ignore.

2. All table methods now check that their arguments have the same client as the table stub itself. This prevents mixing tables from different clients in the same method.